### PR TITLE
fix(Core/Object) Increase WG general visibility and Cannon damage range

### PR DESCRIPTION
**Pull request by Wowhungarian**
## Changes Proposed:
- General visibility at WG should be increased, cause right now, it's not on retail value. Retail value is around 280 yard.
- Fire Cannon's range should be around 300 yard. To pervent invisible Cannon dmg, WG units (Tower Cannons) visibility needs to be greater than 300 yard.

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4113 https://github.com/azerothcore/azerothcore-wotlk/issues/4057

## Tests Performed:
- Built without errors
- Tested on Windows
- Tested ingame, with a pentium cpu, 1050ti gpu, on a wqhd screen, fps was fine

## How to Test the Changes:
- .tele wintergrasp
- .gm on
- find a WG Tower Cannon
- .gm off
- try to shoot mobs from over 150 yard

## Target Branch(es):
- [x] Master

## Todo list:
- [ ] Find a way to increase only the WG gobjects visibility

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by pklloveyou** [comment]
Another RP has been tested

**Comment by FrancescoBorzi** [comment]
@pklloveyou  thanks for testing this. Can you please include more details about what have you tested? as this is quite a generic change:

```
#define MAX_VISIBILITY_DISTANCE     300.0f
```

**Comment by Wowhungarian** [comment]
@FrancescoBorzi Yes, it is, and it had only negative effects, I already reverted it. We only need a higher general wg visibility.

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
well, ok then for me

**Comment by Wowhungarian** [comment]
@FrancescoBorzi Can I ask, that how will be the dbc files updated? Cause there is no point to increase wg visibility without editing Cannon range.

**Comment by Kitzunu** [comment]
Yoy can change the range of spells in Spellmgr

**Comment by Wowhungarian** [comment]
@Kitzunu Cool, but I need a custom value, which is not in SpellRange.dbc by default. I can only use default range indexes in Spellmgr.cpp.

**Comment by Kitzunu** [comment]
How do you know the exact value? Sniffs?

**Comment by Wowhungarian** [comment]
I tried to explain it in the "Changes Proposed" section, but I know, that it has an index, which refers to a lower range, like what is needed. And there isn't any index that refers to 260 yard. We need exactly 260 yard, cause, if it will be higher that that, fe. 300 yard, players will die from invisible Cannon damage.

**Comment by Wowhungarian** [comment]
I see that Cannon radius has been modified in spellmgr:
`// WintergraspCannon - Spell Fire Cannon - EffectRadiusIndex
    ApplySpellFix({ 51422 }, [](SpellEntry* spellInfo)
    {
        spellInfo->EffectRadiusIndex[0] = EFFECT_RADIUS_10_YARDS; // SPELL_EFFECT_SCHOOL_DAMAGE
    });`
If it stays like that, we will need a 270 yard index for the Cannon projectile, cause I calculated with 20 yard radius before.

**Comment by Kitzunu** [comment]
Well how do you know the retsil value is 280y? 

Because I think its highly unlikely there isnt a proper range index for it. I may be wrong, but feel weird. However blizard is weird lol

**Comment by Wowhungarian** [comment]
@Kitzunu I said it's around 280. I don't know it exactly, but based on the origi video from my issue, it's kinda like that. I tested it with sieges, till they appeared exactly, where they did at the video.

**Comment by Kitzunu** [comment]
I will check some things soon

**Comment by Kitzunu** [comment]
Maybe unrelated but on TC:
```cpp
#define MAX_VISIBILITY_DISTANCE         SIZE_OF_GRIDS           // max distance for visible objects
#define SIZE_OF_GRIDS            533.3333f

**Comment by Kitzunu** [comment]
According to https://wowwiki.fandom.com/wiki/Wintergrasp_Tower_Cannon?oldid=1904020 Fire Cannon 10 - 70 yd

**Comment by Wowhungarian** [comment]
It is just the trigger activator spell, which activate trigger missile effect, which has 150 yard range by default. 
https://www.wowhead.com/spell=51421/fire-cannon
I think this range was increased by Blizzard with a wotlk patch.

**Comment by Kitzunu** [comment]
Yea, but the spell that you use to trigger the other has a max range of 70 yards which means the other spell can only fly so far? I might be wrong, but seems very weird if the triggered spell can fly further than what the actual spell you use can?

**Comment by Wowhungarian** [comment]
@Kitzunu Actually, it can fly to 300+ yard, so I really don't know what that 70 yard mean to be. But I'm sure we have nothing to do with that thing, since editing this value changes nothing.

**Comment by Kitzunu** [comment]
Im just very confused

**Comment by pklloveyou** [comment]
Far is better than near

**Comment by Wowhungarian** [comment]
Mark it with WIP pls, I will come up with somethin soon.

File: src/server/game/Entities/Object/Object.cpp [hunk 1]
```diff
@@ -1564,7 +1569,12 @@ float WorldObject::GetSightRange(const WorldObject* target) const
...
             {
                 if (target->IsVisibilityOverridden() && target->GetTypeId() == TYPEID_UNIT)
-                    return MAX_VISIBILITY_DISTANCE;
+                {
+                    if (IsInWintergrasp() && target->IsInWintergrasp())
+                        return MAX_VISIBILITY_FOR_WGUNITS;
+                    else
+                        return MAX_VISIBILITY_DISTANCE;
+                }
                 else if (target->GetTypeId() == TYPEID_GAMEOBJECT)
```
Comment (Wowhungarian) [inline src/server/game/Entities/Object/Object.cpp:1578]: This section is just not working for wg gameobjects.

**Comment by Si1ker** [comment]
@Wowhungarian 👀 ?

**Comment by Wowhungarian** [comment]
Sry guys, I can't solve it.

**Pull request closed without merging**